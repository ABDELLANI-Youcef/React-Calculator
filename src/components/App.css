/* global settings */

*,
*::after,
*::before {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  overflow-x: hidden;
}

html {
  min-width: 22rem;
  min-height: 44rem;
  display: block;
  font-family: 'Raleway', sans-serif;
}

body {
  width: 100%;
  min-height: 100vh;
  min-width: 22rem;
  display: block;
  background-color: #ecc171;
}
