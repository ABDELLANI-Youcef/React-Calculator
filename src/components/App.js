import Display from './Display';
import Buttonspanel from './ButtonsPanel';
import calculate from '../logic/calculate';

function App() {
  return (
    <>
      <Display result="0" />
      <Buttonspanel calculate={calculate} />
    </>
  );
}

export default App;
