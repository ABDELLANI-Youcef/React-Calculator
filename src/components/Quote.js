const Quote = () => (
  <div>Quote</div>
);

export default Quote;
