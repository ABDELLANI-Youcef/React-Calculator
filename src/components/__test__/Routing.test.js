import React from 'react';
import renderer from 'react-test-renderer';
import {
  BrowserRouter,
} from 'react-router-dom';
import App from '../Routes';
import Quote from '../Quote';
import Calculator from '../Calculator';

it('should render the home page of math magicien correctly', () => {
  const home = renderer.create(
    <App />,
  ).toJSON();
  expect(home).toMatchSnapshot();
});

it('should render Quotes page', () => {
  const quote = renderer.create(
    <BrowserRouter>
      <Quote />
    </BrowserRouter>,
  ).toJSON();
  expect(quote).toMatchSnapshot();
});

it('should render Calculator page', () => {
  const calculator = renderer.create(
    <BrowserRouter>
      <Calculator />
    </BrowserRouter>,
  ).toJSON();
  expect(calculator).toMatchSnapshot();
});
