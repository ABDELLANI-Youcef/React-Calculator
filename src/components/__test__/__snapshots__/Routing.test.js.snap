// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Calculator page 1`] = `
Array [
  <div
    className="navbar"
  >
    <div>
      Math Magicians
    </div>
    <div>
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
       |  
      <a
        href="/calculator"
        onClick={[Function]}
      >
        calculator
      </a>
       | 
      <a
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </div>
  </div>,
  <div
    className="display"
  >
    <p>
      0
    </p>
  </div>,
  <div
    className="row"
  >
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "width": "25%",
        }
      }
      type="button"
    >
      AC
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "width": "25%",
        }
      }
      type="button"
    >
      +/-
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "width": "25%",
        }
      }
      type="button"
    >
      %
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
          "width": "25%",
        }
      }
      type="button"
    >
      ÷
    </button>
  </div>,
  <div
    className="row"
  >
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "width": "25%",
        }
      }
      type="button"
    >
      7
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "width": "25%",
        }
      }
      type="button"
    >
      8
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "width": "25%",
        }
      }
      type="button"
    >
      9
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
          "width": "25%",
        }
      }
      type="button"
    >
      X
    </button>
  </div>,
  <div
    className="row"
  >
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "width": "25%",
        }
      }
      type="button"
    >
      4
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "width": "25%",
        }
      }
      type="button"
    >
      5
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "width": "25%",
        }
      }
      type="button"
    >
      6
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
          "width": "25%",
        }
      }
      type="button"
    >
      -
    </button>
  </div>,
  <div
    className="row"
  >
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "width": "25%",
        }
      }
      type="button"
    >
      1
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "width": "25%",
        }
      }
      type="button"
    >
      2
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "width": "25%",
        }
      }
      type="button"
    >
      3
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
          "width": "25%",
        }
      }
      type="button"
    >
      +
    </button>
  </div>,
  <div
    className="row"
  >
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "width": "50%",
        }
      }
      type="button"
    >
      0
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "width": "25%",
        }
      }
      type="button"
    >
      .
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
          "width": "25%",
        }
      }
      type="button"
    >
      =
    </button>
  </div>,
]
`;

exports[`should render Quotes page 1`] = `
<div>
  <div
    className="navbar"
  >
    <div>
      Math Magicians
    </div>
    <div>
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
       |  
      <a
        href="/calculator"
        onClick={[Function]}
      >
        calculator
      </a>
       | 
      <a
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </div>
  </div>
  <p>
    Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding. –William Paul Thurston
  </p>
</div>
`;

exports[`should render the home page of math magicien correctly 1`] = `
<div>
  <div>
    <div
      className="navbar"
    >
      <div>
        Math Magicians
      </div>
      <div>
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
         |  
        <a
          href="/calculator"
          onClick={[Function]}
        >
          calculator
        </a>
         | 
        <a
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </div>
    </div>
    <h1>
      Welcome to my page!
    </h1>
    <p>
      This is my React project that aims to make a simple calculator.
    </p>
    <p>
      It is created using React and lets you make the next operations: addition, substraction, multiplication, division and percentage
    </p>
  </div>
</div>
`;
